### ISSUE TITLE
how to configure problemmatcher

### ISSUE THREAD

**Issue by d-chris** [microsoft/vscode-makefile-tools#255]
_how can i configure the problemmatcher used with makefile-tools?_

clicking on the "problem" will result in following error due a relative path which is generated from the output of the avr-gcc

## problem
`Unable to open '3450041.c': Unable to read file '\3450041.c' (Error: Unable to resolve nonexistent file '\3450041.c').`

## terminal
```
make.exe: Entering directory `D:/SVN/WORK/3450041/cavr/PeugeotTS'

-------- begin --------
avr-gcc (GCC) 3.4.6
Copyright (C) 2006 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO 
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


Compiling: 3450041.c
avr-gcc -c -mmcu=atmega8 -I. -gdwarf-2 -DCAN_CLK16M -DHARDWARE_V1_2 -DF_CPU=16000000U -I../shared -I../avrlib  -Os -funsigned-char -funsigned-bitfields -fpack-struct -fshort-enums -Wall -Wstrict-prototypes -Wa,-adhlns=3450041.lst  -std=gnu99 -Wp,-M,-MP,-MT,3450041.o,-MF,.dep/3450041.o.d 3450041.c -o 3450041.o 
3450041.c: In function `main':
3450041.c:132: error: syntax error before "InitializeCanNode"
make.exe: *** [3450041.o] Error 1    
make.exe: Leaving directory `D:/SVN/WORK/3450041/cavr/PeugeotTS'
The terminal process "C:\WINDOWS\System32\cmd.exe /d /c ""make.exe" "all" "-C" "./cavr/PeugeotTS""" terminated with exit code: 2.

Terminal will be reused by tasks, press any key to close it.
```

## settings.json
```json 
{
    "makefile.extensionOutputFolder": "./.vscode",
    "makefile.preConfigureScript": "./.vscode/winavr346.bat", // load path from make.exe into PATH-env
    "makefile.alwaysPreConfigure": true,
    "makefile.clearOutputBeforeBuild": true,
    "makefile.configurations": [
        {
            "name": "PeugeotTS",
            "makeDirectory": "./cavr/PeugeotTS",
            "problemMatchers": ["$gcc"]
        },
    ],
    "files.exclude": {
        "**/{CVS,.dep}": true, // hide directories
        "**/*.{o,d,lst}": true, // hide object files
        "**/*.{map,lss,sym}": true, // hide compiler files
        "**/*.{elf,eep}": true, // hide binary files
        "**/*.hex": true // hide firmware
    },
    "C_Cpp.default.includePath": [
        "${workspaceFolder}/**",
        "${env:WINAVR}/avr/include/" // include path for standard-header
    ],
    "C_Cpp.default.configurationProvider": "ms-vscode.makefile-tools"
}
```

**Comment by andreeis** [comment]
@d-chris, thank you for creating this bug report. I suspect what the problem is and to verify my theory, can you open your project from ./cavr/PeugeotTS folder (and don't use "makefile.configurations[].makeDirectory") and if you build from there, does the problem matcher work?

The source file 3450041.c is in PeugeotTS folder?

**Comment by andreeis** [comment]
Also, is there a way to make avr-gcc compiler log full paths? That should unblock you until we fix the problem (which is we pass ${workspaceFolder} as a base for relative paths analysis instead of "makefile.makeDirectory").

**Comment by d-chris** [comment]
> Also, is there a way to make avr-gcc compiler log full paths? That should unblock you until we fix the problem (which is we pass ${workspaceFolder} as a base for relative paths analysis instead of "makefile.makeDirectory").

@andreeis  for testing i changed the path in my `makefile` to an absolute path. because i don't know if there is a compiler-option for that.
 => it is working,

**Comment by andreeis** [comment]
@d-chris, thank you for running the experiments and confirming my theory.
One part of the problem we will fix now with the upcoming 0.4 release. I already have a prototype working. We will pass any folder we find given with -C or -f (extract folder from full makefile path) to the problem matcher so that it doesn't try workspace folder.

If your build will move through directories more than once, the task will be able to see only the first directory.
With workitem https://github.com/microsoft/vscode-makefile-tools/issues/257, we will try to support also more complicated scenarios of make jumping and building all around.

**Comment by andreeis** [comment]
@d-chris, we released Makefile Tools 0.4.0 today which contains the partial fix we discussed above and also worth noting, localization support. If you happen to use VSCode language packs, give Makefile Tools a try in your preferred language.
Let us know if you encounter any issues: with the translated strings, any other bugs or if you have any feedback for the future of this extension.

**Comment by d-chris** [comment]
> @d-chris, we released Makefile Tools 0.4.0 today which contains the partial fix we discussed above and also worth noting, localization support. If you happen to use VSCode language packs, give Makefile Tools a try in your preferred language.
> 
> Let us know if you encounter any issues: with the translated strings, any other bugs or if you have any feedback for the future of this extension.

Will do some testing...
shall I test for Bavarian accent or proper German? 🤪
